foreshadow/preprocessor.py

```diff
@@ -141,14 +148,14 @@ def _map_pipelines(self):
             },
             # Extracts already resolved single pipelines from JSON intent overrides
             **{
-                k: self._intent_pipelines[v.__name__].get(
+                k: deepcopy(self._intent_pipelines[v.__name__].get(
                     "single",
                     Pipeline(
                         deepcopy(v.single_pipeline(self.y_var))
```

Comment:
Since we are deep copying at the upper level we likely don't need this here since only the value matters (not the object reference itself).